src/ansiblelint/rules/var_naming.py

```diff
@@ -46,44 +46,44 @@ def get_var_naming_matcherror(
                 message="Variables names must be strings.",
                 rule=self,
             )
+        if ident not in ANNOTATION_KEYS:
```

Comment:
If you reverse the condition you can just return None fast and avoid adding a new level of indentation to this function. Basically we already know that we don't need to do anything if the key (ident) is one of these.